covfee/shared/dataclass.py

```diff
@@ -61,16 +61,39 @@ def __post_init__(self):
 
 
 class Journey(BaseDataclass):
-    def __init__(self, nodes: List[CovfeeTask] = None, name: str = None):
+    name: str
+    global_unique_id: Optional[str]
+    nodes_players: List[Tuple[CovfeeTask, int]]
```

Comment:
I don't think this variables should be defined here, if you want type hints, why not add them in the init?

Reply:
I'm fine with having it like but to be honest, I don't like the subtle change in semantics where something is a class variable if it is initialised, i.e.
```python
class MyClass:
  var_1: int # Instance var
  var_2: int = 2 # Class var
  var_3: ClassVar[int] = 2 # Class var
  var_4 = 2 # Class var
  var_5: ClassVar[int] # throws error?
```
in particular, the difference between `var_1` and `var_2`. Anyhow, I like being PEP compliant, so happy to adapt to this.

Reply:
Alright, but the PEP only annotates variables as class variables when the explicitly use `ClassVar`. Whether an instance var is initialized or not, just introduces the assumption that non initialized variables are meant to be assigned some value in __ini__

![image](https://github.com/TUDelft-SPC-Lab/covfee/assets/157802575/59c6dbfa-d125-4366-a7bf-6cb5c98cec22)

But thanks for approving.

Reply:
Yes, I read the whole PEP, but I mean that in practice because it's backwards compatible, all the options that I wrote are still valid and do happen unless we make all the code PEP compliant.